diam/sm/smparser/app.go

```diff
@@ -19,9 +19,34 @@ type Application struct {
 	id                          []uint32 // List of supported application IDs.
 }
 
+// SupportedApp holds properties of each locally supported App
+type SupportedApp struct {
+	ID      uint32
+	AppType string
+	Vendor  uint32
+}
+
+// PrepareSupportedApps prepares a list of locally supported apps
+func PrepareSupportedApps() []*SupportedApp {
+	locallySupportedApps := []*SupportedApp{}
+	for _, app := range dict.Default.Apps() {
```

Comment:
i don't think we should assume that `dict.Default.Apps()` is used here. It should probably be a parameter to `PrepareSupportedApps`.